plugins/server.go

```diff
@@ -49,9 +49,18 @@ type Server struct {
 }
 
 func newServer(loggerName string) (*Server, error) {
+	envCfg, err := GetEnvConfig()
+	if err != nil {
+		return nil, fmt.Errorf("error getting environment configuration: %w", err)
+	}
```

Comment:
Can we avoid calling this twice? We had a clean separation between creation and start before that is becoming muddled here. Since the `TracingConfig` doesn't affect the `GRPCOpts`, separating those responsibilities as two funcs would let us drop this by moving the `TracingConfig` to `*Server.start()`.